scripts/extensions/videoEditor/src/VideoTimeline/ListThumbnails.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import VideoEditorContext from '../VideoEditorContext';
+import {IThumbnail} from '../interfaces';
+
+interface IProps {
+    thumbnails: Array<IThumbnail>;
+}
+
+export class ListThumbnails extends React.PureComponent<IProps> {
+    static contextType = VideoEditorContext;
+
+    render() {
+        const {getClass} = this.context.superdesk.utilities.CSS;
+
+        return (
+            <div className={`${getClass('frames')} ${getClass('frames--thumbs')}`}>
+                <div className={getClass('frames__inner')}>
+                    {this.props.thumbnails.map((item: IThumbnail, index: number) => (
+                        <video
```

Comment:
We use `video` element. Because it support get thumbnails directly in video stream(via time). But it very slow. it will faster if we use `blob` (but it must pre-download file to browser)
So we change to use solution: pre-capture 50 pics in server. We keep video element because we want to find another way to make it faster in future.
But we think we should change to `img` element for now. Maybe we will update again to video later if we find any way better.

Reply:
We use `video` element. Because it support get thumbnails directly in video stream(via time). But it very slow. it will faster if we use `blob` (but it must pre-download file to browser)
So we change to use solution: pre-capture 50 pics in server. We keep video element because we want to find another way to make it faster in future.
But we think we should change to `img` element for now. Maybe we will update again to video later if we find any way better.

Reply:
Thanks for explaining. Generating thumbnails on the server is technically simpler and will probably always be faster since you can do it as soon as the video is uploaded while if you wanted to do it on the client you have to download parts of the video which probably doesn't work in parallel either.